src/services/webclient.ts

```diff
@@ -137,7 +137,8 @@ export class WebClientService implements threema.WebClientService {
     // State handling
     private startupPromise: ng.IDeferred<{}>; // TODO: deferred type
     private startupDone: boolean = false;
-    private initialized = 0;
+    private pendingInitializationStepRoutines: threema.InitializationStepRoutine[] = [];
+    private initialized: threema.InitializationStep[] = [];
```

Comment:
To make sure that no step is added twice, you could make this a `Set<threema.InitializationStep>`: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set You'd have to replace any `indexOf` call with `has` and any call to `push` with `add`.